Mage.Sets/src/mage/cards/b/BoromirWardenOfTheTower.java

```diff
@@ -0,0 +1,103 @@
+package mage.cards.b;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.costs.common.SacrificeSourceCost;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.CounterTargetEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAllEffect;
+import mage.abilities.effects.keyword.TheRingTemptsYouEffect;
+import mage.abilities.keyword.IndestructibleAbility;
+import mage.abilities.keyword.VigilanceAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.stack.Spell;
+import mage.target.targetpointer.FixedTarget;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author Susucr
+ */
+public final class BoromirWardenOfTheTower extends CardImpl {
+
+    public BoromirWardenOfTheTower(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{W}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SOLDIER);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Vigilance
+        this.addAbility(VigilanceAbility.getInstance());
+
+        // Whenever an opponent casts a spell, if no mana was spent to cast it, counter that spell.
+        this.addAbility(new BoromirWardenOfTheTowerTriggeredAbility());
```

Comment:
Possible to factor out from this and Lavinia to common ability class?

Reply:
Sure,
my usual rule of thumb is to refactor when there are 3 instances of the same code piece, so did not bother making a class for that trigger.